### PR TITLE
add deprecation docs for PostgresSQL/MySQL targets

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
add deprecation docs for PostgresSQL/MySQL targets

## Motivation and Context
doc update

## How to test this PR?
As mentioned in the docs

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [x] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#c98f23c2-b972-4529-9491-8ab2106a2487) |
<a name="c98f23c2-b972-4529-9491-8ab2106a2487"></a>
### 9333-e0331c9/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:31769
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 18f36d87c708:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 7 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 1 minutes and 13 seconds
(4/15) Running aws-sdk-ruby tests ... done in 15 seconds
(5/15) Running awscli tests ... done in 3 minutes and 0 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 6 minutes and 11 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 31 seconds
(9/15) Running minio-go tests ... done in 20 minutes and 49 seconds
(10/15) Running minio-java tests ... done in 27 minutes and 8 seconds
(11/15) Running minio-js tests ... FAILED in 4 minutes and 3 seconds
{
  "name": "minio-js",
  "function": "listObjectsV2WithMetadata(bucketName, prefix, recursive, startAfter)",
  "args": "bucketName:minio-js-test-896ea006-ea05-40fc-8dc9-1217a25836b7, recursive:true",
  "duration": 209,
  "status": "FAIL",
  "error": "TypeError: Cannot read property '0' of undefined at /mint/run/core/minio-js/node_modules/minio/dist/main/xml-parsers.js:414:42 at Array.forEach (<anonymous>) at parseListObjectsV2WithMetadata (node_modules/minio/dist/main/xml-parsers.js:409:21) at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:88:19) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}

Executed 10 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#badd1256-f177-4924-97c3-2cc1c41b4cfe) |
<a name="badd1256-f177-4924-97c3-2cc1c41b4cfe"></a>
### 9333-b2316c3/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:32497
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 1cd35f1ae40c:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 1 minutes and 14 seconds
(4/15) Running aws-sdk-ruby tests ... done in 14 seconds
(5/15) Running awscli tests ... done in 2 minutes and 33 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 6 minutes and 21 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 31 seconds
(9/15) Running minio-go tests ... done in 18 minutes and 57 seconds
(10/15) Running minio-java tests ... done in 27 minutes and 34 seconds
(11/15) Running minio-js tests ... FAILED in 4 minutes and 4 seconds
{
  "name": "minio-js",
  "function": "listObjectsV2WithMetadata(bucketName, prefix, recursive, startAfter)",
  "args": "bucketName:minio-js-test-712c7468-2a31-42b2-aa47-408cd64fe41d, recursive:true",
  "duration": 213,
  "status": "FAIL",
  "error": "TypeError: Cannot read property '0' of undefined at /mint/run/core/minio-js/node_modules/minio/dist/main/xml-parsers.js:414:42 at Array.forEach (<anonymous>) at parseListObjectsV2WithMetadata (node_modules/minio/dist/main/xml-parsers.js:409:21) at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:88:19) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}

Executed 10 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by cesnietor — APPROVED** [review]

File: docs/bucket/notifications/README.md [hunk 1]
```diff
@@ -807,6 +807,28 @@ Received a message: {"EventType":"s3:ObjectCreated:Put","Key":"images/myphoto.jp
...
+> password            (string)             database password (used only if `connection_string` is empty)
+> database            (string)             database name (used only if `connection_string` is empty)
+> ```
+>
+> These are now deprecated, if you plan to upgrade to any releases after *RELEASE.2020-04-10T03-34-42Z* please make
+> sure to migration to only using *connection_string* option. To migrate, once you have upgraded all your servers
+> use the following command to update your notification targets.
+>
+> ```
+> mc admin config set myminio/ notifiy_postgres[:name] connection_string="host=hostname port=2832 username=psqluser password=psqlpass database=bucketevents"
```
Comment (poornas) [inline docs/bucket/notifications/README.md]:
```suggestion
> mc admin config set myminio/ notify_postgres[:name] connection_string="host=hostname port=2832 username=psqluser password=psqlpass database=bucketevents"
```

File: docs/bucket/notifications/README.md [hunk 2]
```diff
@@ -807,6 +807,28 @@ Received a message: {"EventType":"s3:ObjectCreated:Put","Key":"images/myphoto.jp
...
+> ```
+> host                (hostname)           Postgres server hostname (used only if `connection_string` is empty)
+> port                (port)               Postgres server port, defaults to `5432` (used only if `connection_string` is empty)
+> username            (string)             database username (used only if `connection_string` is empty)
+> password            (string)             database password (used only if `connection_string` is empty)
+> database            (string)             database name (used only if `connection_string` is empty)
+> ```
+>
+> These are now deprecated, if you plan to upgrade to any releases after *RELEASE.2020-04-10T03-34-42Z* please make
+> sure to migration to only using *connection_string* option. To migrate, once you have upgraded all your servers
```
Comment (poornas) [inline docs/bucket/notifications/README.md]:
```suggestion
> These are now deprecated, if you plan to upgrade to any releases after *RELEASE.2020-04-10T03-34-42Z* please make
> sure to migrate using *connection_string* option only. To migrate, once you have upgraded all your servers
```

File: docs/bucket/notifications/README.md [hunk 3]
```diff
@@ -914,6 +936,28 @@ key                 |                      value
...
+> ```
+> host         (hostname)           MySQL server hostname (used only if `dsn_string` is empty)
+> port         (port)               MySQL server port (used only if `dsn_string` is empty)
+> username     (string)             database username (used only if `dsn_string` is empty)
+> password     (string)             database password (used only if `dsn_string` is empty)
+> database     (string)             database name (used only if `dsn_string` is empty)
+> ```
+>
+> These are now deprecated, if you plan to upgrade to any releases after *RELEASE.2020-04-10T03-34-42Z* please make
+> sure to migration to only using *dsn_string* option. To migrate, once you have upgraded all your servers
```
Comment (poornas) [inline docs/bucket/notifications/README.md]:
```suggestion
> These are now deprecated, if you plan to upgrade to any releases after *RELEASE.2020-04-10T03-34-42Z* please make
> sure to migrate using *dsn_string* option only. To migrate, once you have upgraded all your servers
```

File: docs/bucket/notifications/README.md [hunk 4]
```diff
@@ -914,6 +936,28 @@ key                 |                      value
...
+> use the following command to update your notification targets.
+>
+> ```
+> mc admin config set myminio/ notifiy_mysql[:name] dsn_string="mysqluser:mysqlpass@tcp(localhost:2832)/bucketevents"
```
Comment (poornas) [inline docs/bucket/notifications/README.md]:
```suggestion
> mc admin config set myminio/ notify_mysql[:name] dsn_string="mysqluser:mysqlpass@tcp(localhost:2832)/bucketevents"
```

**Review by poornas — CHANGES_REQUESTED** [review]

**Review by poornas — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#3cb0cfa7-43d4-44e5-b973-1c451de9f6fe) |
<a name="3cb0cfa7-43d4-44e5-b973-1c451de9f6fe"></a>
### 9333-d9d5536/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:31209
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f3e1d225ba54:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 7 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 1 minutes and 21 seconds
(4/15) Running aws-sdk-ruby tests ... done in 15 seconds
(5/15) Running awscli tests ... done in 3 minutes and 10 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 6 minutes and 21 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 27 seconds
(9/15) Running minio-go tests ... done in 19 minutes and 27 seconds
(10/15) Running minio-java tests ... done in 27 minutes and 7 seconds
(11/15) Running minio-js tests ... FAILED in 4 minutes and 1 seconds
{
  "name": "minio-js",
  "function": "listObjectsV2WithMetadata(bucketName, prefix, recursive, startAfter)",
  "args": "bucketName:minio-js-test-9a727dd7-40c0-412e-931f-e86d476791e9, recursive:true",
  "duration": 211,
  "status": "FAIL",
  "error": "TypeError: Cannot read property '0' of undefined at /mint/run/core/minio-js/node_modules/minio/dist/main/xml-parsers.js:414:42 at Array.forEach (<anonymous>) at parseListObjectsV2WithMetadata (node_modules/minio/dist/main/xml-parsers.js:409:21) at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:88:19) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}

Executed 10 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Pull request merged**